### PR TITLE
Tests for exporter (regression for #746) + some test refactoring

### LINKED ISSUE #685: Test exporter
There should be a test making sure that exporting a semester downloads a file without errors.

### PR DESCRIPTION
**Pull request by schumannd**
Fixes #685 

Refactoring also a part of ongoing test reorganization #707

### CODE REVIEW

**Comment by schumannd** [comment]
I had to rename staff/tests.py as this would conflict with the new staff/tests folder. It seems that landscape just now sees the code smells in that file. Maybe it does not check any "tests" folders or files? In that case I could move tests_general.py to the test folder as well.

**Comment by janno42** [comment]
i have no idea whether landscape ignores some files or folders. the docs are not very helpful.
is there any reason for not moving the `tests_general.py` into the tests folder?

File: evap/evaluation/tests/test_utils.py [hunk 1]
```diff
@@ -9,3 +12,28 @@ def test_check_response_code_200(self):
...
+
+    if id:
+        course = mommy.make(Course, state='prepared', id=id)
+    else:
+        course = mommy.make(Course, state='prepared')
+
+    mommy.make(Contribution, course=course, contributor=contributor, can_edit=True, responsible=True)
+    mommy.make(Contribution, course=course, contributor=editor, can_edit=True)
+
+    return course
```
Comment (janno42) [inline evap/evaluation/tests/test_utils.py]: please add newline at end of file

File: evap/contributor/tests/test_views.py [hunk 2]
```diff
@@ -43,7 +29,7 @@ class TestContributorCourseView(ViewTest):
 
     @classmethod
     def setUpTestData(cls):
-        course_with_responsible_and_editor()
+        course_with_responsible_and_editor(2)
```
Comment (janno42) [inline evap/contributor/tests/test_views.py]: please use `course_with_responsible_and_editor(course_id=2)` here and two more times below to make clear what the argument is for.

File: evap/evaluation/tests/test_utils.py [hunk 3]
```diff
@@ -9,3 +12,28 @@ def test_check_response_code_200(self):
...
+    return page.forms[max(key for key in page.forms.keys() if isinstance(key, int))]
+
+
+def get_form_data_from_instance(FormClass, instance):
+    assert FormClass._meta.model == type(instance)
+    form = FormClass(instance=instance)
+    return {field.html_name: field.value() for field in form}
+
+
+def course_with_responsible_and_editor(id=None):
```
Comment (janno42) [inline evap/evaluation/tests/test_utils.py]: please use `course_id` as argument

**Comment by schumannd** [comment]
Added newline, moved file and rebased branch. Should be good to go now.

File: evap/staff/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+from evap.evaluation.models import Semester, UserProfile, Course, CourseType
+from evap.evaluation.tests.test_utils import lastform, ViewTest
+
+
+class TestSemesterExportView(ViewTest):
+    url = '/staff/semester/1/export'
+    users = ['staff']
+
+    def setUp(self):
+        self.semester = mommy.make(Semester, name='fuk', name_en='fuk', name_de='fuk')
```
Comment (janno42) [inline evap/staff/tests/test_views.py]: can you remove the names here and for the coursetype below and let mommy create the names?

File: evap/staff/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        general_contribution = mommy.make(Contribution, course=course, contributor=None)
+        course = Course.objects.get()  # refresh because of cached properties
+        self.assertFalse(course.has_enough_questionnaires())
+
+        q = mommy.make(Questionnaire)
+        general_contribution.questionnaires.add(q)
+        self.assertFalse(course.has_enough_questionnaires())
+
+        responsible_contribution.questionnaires.add(q)
+        self.assertTrue(course.has_enough_questionnaires())
```
Comment (janno42) [inline evap/staff/tests/test_models.py]: newline

File: evap/staff/tests/test_views.py (hunk 4 above)
Comment (schumannd) [inline evap/staff/tests/test_views.py]: haha, oops. That are the results of some frustrating debugging there...

**Pull request merged**